owncloudApp/src/main/java/com/owncloud/android/ui/activity/ShareActivity.kt

```diff
@@ -0,0 +1,394 @@
+import com.owncloud.android.lib.resources.shares.RemoteShare
+import com.owncloud.android.lib.resources.shares.ShareParserResult
+import com.owncloud.android.lib.resources.shares.ShareType
+import com.owncloud.android.operations.*
+import com.owncloud.android.providers.UsersAndGroupsSearchProvider
+import com.owncloud.android.shares.db.OCShare
+import com.owncloud.android.ui.asynctasks.GetSharesForFileAsyncTask
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import com.owncloud.android.ui.fragment.*
+
+
+/**
+ * Activity for sharing files
+ */
+class ShareActivity : FileActivity(), ShareFragmentListener {
+
+    internal var mGetSharesForFileAsyncTask: GetSharesForFileAsyncTask? = null
+
+    /**
+     * Shortcut to get access to the [ShareFileFragment] instance, if any
+     *
+     * @return A [ShareFileFragment] instance, or null
+     */
+    private val shareFileFragment: ShareFileFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_SHARE_FRAGMENT) as ShareFileFragment
+
+    /**
+     * Shortcut to get access to the [SearchShareesFragment] instance, if any
+     *
+     * @return A [SearchShareesFragment] instance, or null
+     */
+    private val searchFragment: SearchShareesFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_SEARCH_FRAGMENT) as SearchShareesFragment
+
+    /**
+     * Shortcut to get access to the [PublicShareDialogFragment] instance, if any
+     *
+     * @return A [PublicShareDialogFragment] instance, or null
+     */
+    private val publicShareFragment: PublicShareDialogFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_PUBLIC_SHARE_DIALOG_FRAGMENT) as PublicShareDialogFragment
+
+    /**
+     * Shortcut to get access to the [EditShareFragment] instance, if any
+     *
+     * @return A [EditShareFragment] instance, or null
+     */
+    private val editShareFragment: EditShareFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_EDIT_SHARE_FRAGMENT) as EditShareFragment
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mGetSharesForFileAsyncTask = null
+
+        setContentView(R.layout.share_activity)
+
+        // Set back button
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+
+        val ft = supportFragmentManager.beginTransaction()
+
+        if (savedInstanceState == null) {
+            // Add Share fragment on first creation
+            val fragment = ShareFileFragment.newInstance(file, account)
+            ft.replace(R.id.share_fragment_container, fragment, TAG_SHARE_FRAGMENT)
+            ft.commit()
+        }
+
+    }
+
+    override fun onAccountSet(stateWasRecovered: Boolean) {
+        super.onAccountSet(stateWasRecovered)
+        // Load data into the list
+        Log_OC.d(TAG, "Refreshing lists on account set");
+
+        // TODO New Android Components
+//        refreshSharesFromStorageManager();
```

Comment:
can you delete commented code here and a couple of lines below?

Reply:
done